src/Evolution/Systems/CurvedScalarWave/BoundaryConditions/Worldtube.hpp

```diff
@@ -0,0 +1,146 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/BoundaryConditions/Type.hpp"
+#include "Evolution/Systems/CurvedScalarWave/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace CurvedScalarWave::BoundaryConditions {
+
+/*!
+ * \brief Sets boundary conditions for the elements abutting the worldtube using
+ * a combination of constraint-preserving boundary conditions and the local
+ * solution evolved inside the worldtube.
+ *
+ * \details After extensive experimentation we found this set of boundary
+ * conditions to be optimal. They are formulated in terms of characteristic
+ * fields.
+ *
+ * Boundary conditions for \f$w^0_\psi\f$ \f$w^0_i\f$ are formulated
+ * by demanding that there are no constraint violations flowing into the
+ * numerical domain and are formulated as corrections to the time derivative of
+ * the evolved variables directly. The derivation is described in
+ * \ref ConstraintPreservingSphericalRadiation .
+ *
+ * Boundary conditions for \f$\w^-\f$ are formulated by evaluating the
+ * analytical solution of the worldtube at the grid points of each element
+ * abutting the worldtube. The data is updated and saved to
+ * \ref CurvedScalarWave::Worldtube::Tags::WorldtubeSolution each time step.
+ * It is then treated like a ghost field and applied with the chosen numerical
+ * flux. So far only the upwind flux has been tried.
+ *
+ * We tried several other combinations such as setting all fields from the
```

Comment:
These are really great things to have in the documentation. Thank you for adding them!